DFM/dfmSession.c

```diff
@@ -135,10 +135,11 @@ DfmResult_t xDfmSessionEnable(uint32_t ulOverride)
 	if ((ulStoredEnabledValue == DFM_SESSION_STORAGE_DUMMY_VALUE) || ((ulOverride == 1UL) && (ulStoredEnabledValue == DFM_DISABLED)))
 	{
 		/* Couldn't read stored value or we override a disabled value */
-		((DfmSessionStorage_t*)cSessionStorageBuffer)->ulVersion = DFM_SESSION_STORAGE_VERSION; /*cstat !MISRAC2012-Rule-11.3 We use an untyped buffer to retrieve the session data since we can't know what version and size it might have been stored in the past. The stored Session data might be larger than the current DfmSessionStorage_t*/
-		((DfmSessionStorage_t*)cSessionStorageBuffer)->ulEnabled = DFM_ENABLED; /*cstat !MISRAC2012-Rule-11.3 We use an untyped buffer to retrieve the session data since we can't know what version and size it might have been stored in the past. The stored Session data might be larger than the current DfmSessionStorage_t*/
+		DfmSessionStorage_t *pSessionStorageBuffer = (DfmSessionStorage_t*)&cSessionStorageBuffer[0];
```

Comment:
Please rename this to pxSessionStorage instead.

Reply:
Fixed